ethereum/core/src/test/java/org/hyperledger/besu/ethereum/worldstate/StateTrieAccountValueTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.worldstate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.rlp.RLP;
+import org.hyperledger.besu.ethereum.rlp.RLPInput;
+import org.hyperledger.besu.util.bytes.BytesValue;
+
+import org.junit.Test;
+
+public class StateTrieAccountValueTest {
+
+  @Test
+  public void roundTripMainNetAccountValueVersionZero() {
+    final long nonce = 0;
+    final Wei balance = Wei.ZERO;
+    // Have the storageRoot and codeHash as different values to ensure the encode / decode
+    // doesn't cross the values over.
+    final Hash storageRoot = Hash.EMPTY_TRIE_HASH;
```

Comment:
Do we want any tests that have non `EMPTY` values?

Reply:
An Empty Trie Hash is the message digest of an empty Trie. It is not zero. All I am doing here is ensuring there is a different value in storageRoot to the codeHash, to make sure we aren't somehow copying two values over each other.

I don't think this is a problem.